<a href="https://colab.research.google.com/github/konkasivaji/690-stats-and-visulaization/blob/main/world_development_explorer/Part-B/Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Public Sector
The public sector is also part of the economy that maintains infrastructure, public transportation, public education, health care, and police and military services, among other things. In today's world, every nation devotes a greater percentage of its GDP on the Military. Military has a crucial role in protecting democratic and free market movements from both internal and external dangers by providing security assistance, training, and equipment. So lets investigate the publis sector. with this information, like based on its military spending and level of dependence on foreign countries when it comes to imports.

## Why is it important to you and/or to others?
Spending on the military makes the military stronger. This power makes people less likely to attack on country. If the country doesn't spend money on the military, its enemies may see it as an easy target. So, even though local defense spending may be more expensive, it creates more value in the form of stronger national security and less disruptions to trade and commerce. If a country's defense system is fully integrated, it can also help the economy.


## What questions do you have in mind and would like to answer?
Based on these indicators **Military expendicture (cureent USD), Military expenditure(% of GDP) and Arms imports( SIPRI trend indicator values)** i would like to do some research on: 
  * [what percentage of GDP is being spent on military?](#theme-installion)
  * [What percentage of Arms imports are Going on?](#theme-installion)
  * [What are the Military expendicture for those countries?](#theme-installion)

#Analysis
* Data Source : Wang, C. (2021). The world development explorer Available from [http://www.worlddev.xyz](#theme-installion)
* **Topics : Public Sector**
* **[Countries: India, Pakisthan, Russian Federation, China, United States.](#theme-installion)**
* **[Time Period : 2007 - 2020](#theme-installion)**

In [98]:
#Importing required Libraries
import pandas as pd
import plotly
import plotly.express as px

In [99]:
#Reading the data
Data_url = 'https://raw.githubusercontent.com/konkasivaji/690-stats-and-visulaization/main/world_development_explorer/Part-B/wdx_data.csv'

df = pd.read_csv(Data_url)

In [100]:
df.shape

(1481, 10)

In [101]:
df.head()

Unnamed: 0  Year Country Code          Country Name  \
0           0  2007          AFG           Afghanistan   
1           1  2007          AGO                Angola   
2           2  2007          ALB               Albania   
3           3  2007          ARE  United Arab Emirates   
4           4  2007          ARG             Argentina   

                       Region         Income Group    Lending Type  \
0                  South Asia           Low income             IDA   
1         Sub-Saharan Africa   Lower middle income            IBRD   
2       Europe & Central Asia  Upper middle income            IBRD   
3  Middle East & North Africa          High income  Not classified   
4  Latin America & Caribbean   Upper middle income            IBRD   

   MS.MIL.MPRT.KD  MS.MIL.XPND.CD  MS.MIL.XPND.GD.ZS  
0        41000000    2.195802e+08           2.566267  
1        31000000    2.032433e+09           3.114054  
2         5000000    1.948403e+08           1.820765  
3       929000000    8.460994e+09           3.280521  
4        24000000    2.296448e+09           0.792548

In [102]:
df.tail()

Unnamed: 0  Year Country Code   Country Name  \
1476        2275  2020          UKR        Ukraine   
1477        2276  2020          URY        Uruguay   
1478        2277  2020          USA  United States   
1479        2281  2020          XKX         Kosovo   
1480        2283  2020          ZMB         Zambia   

                          Region         Income Group    Lending Type  \
1476       Europe & Central Asia  Lower middle income            IBRD   
1477  Latin America & Caribbean           High income            IBRD   
1478               North America          High income  Not classified   
1479       Europe & Central Asia  Upper middle income             IDA   
1480         Sub-Saharan Africa            Low income             IDA   

      MS.MIL.MPRT.KD  MS.MIL.XPND.CD  MS.MIL.XPND.GD.ZS  
1476        18000000    5.924201e+09           4.129198  
1477        25000000    1.163608e+09           2.321460  
1478       687000000    7.782320e+11           3.741160  
1479         1000000    7.896501e+07           1.060553  
1480         1000000    2.121424e+08           1.154225

In [103]:
df.sample(5)

Unnamed: 0  Year Country Code Country Name                 Region  \
103          165  2008          AUS    Australia    East Asia & Pacific   
1351        2065  2019          NER        Niger    Sub-Saharan Africa    
1045        1608  2016          SWE       Sweden  Europe & Central Asia   
1375        2101  2019          TCD         Chad    Sub-Saharan Africa    
1370        2089  2019          SGP    Singapore    East Asia & Pacific   

     Income Group    Lending Type  MS.MIL.MPRT.KD  MS.MIL.XPND.CD  \
103   High income  Not classified       444000000    1.863309e+10   
1351   Low income             IDA         4000000    2.476321e+08   
1045  High income  Not classified        70000000    5.427549e+09   
1375   Low income             IDA         1000000    2.347097e+08   
1370  High income  Not classified       636000000    1.066245e+10   

      MS.MIL.XPND.GD.ZS  
103            1.798547  
1351           2.651044  
1045           1.053016  
1375           2.187153  
1370           2.865347

In [104]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1481 entries, 0 to 1480
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         1481 non-null   int64  
 1   Year               1481 non-null   int64  
 2   Country Code       1481 non-null   object 
 3   Country Name       1481 non-null   object 
 4   Region             1481 non-null   object 
 5   Income Group       1481 non-null   object 
 6   Lending Type       1481 non-null   object 
 7   MS.MIL.MPRT.KD     1481 non-null   int64  
 8   MS.MIL.XPND.CD     1481 non-null   float64
 9   MS.MIL.XPND.GD.ZS  1481 non-null   float64
dtypes: float64(2), int64(3), object(5)
memory usage: 115.8+ KB


In [105]:
#Re-naming the Indicators
df = df.rename(columns = {"MS.MIL.XPND.GD.ZS":"Military expenditure (% of GDP)","MS.MIL.XPND.CD":"Military expenditure (current USD)","MS.MIL.MPRT.KD":"Arms imports (SIPRI trend indicator values)"})
df

Unnamed: 0  Year Country Code          Country Name  \
0              0  2007          AFG           Afghanistan   
1              1  2007          AGO                Angola   
2              2  2007          ALB               Albania   
3              3  2007          ARE  United Arab Emirates   
4              4  2007          ARG             Argentina   
...          ...   ...          ...                   ...   
1476        2275  2020          UKR               Ukraine   
1477        2276  2020          URY               Uruguay   
1478        2277  2020          USA         United States   
1479        2281  2020          XKX                Kosovo   
1480        2283  2020          ZMB                Zambia   

                          Region         Income Group    Lending Type  \
0                     South Asia           Low income             IDA   
1            Sub-Saharan Africa   Lower middle income            IBRD   
2          Europe & Central Asia  Upper middle income            IBRD   
3     Middle East & North Africa          High income  Not classified   
4     Latin America & Caribbean   Upper middle income            IBRD   
...                          ...                  ...             ...   
1476       Europe & Central Asia  Lower middle income            IBRD   
1477  Latin America & Caribbean           High income            IBRD   
1478               North America          High income  Not classified   
1479       Europe & Central Asia  Upper middle income             IDA   
1480         Sub-Saharan Africa            Low income             IDA   

      Arms imports (SIPRI trend indicator values)  \
0                                        41000000   
1                                        31000000   
2                                         5000000   
3                                       929000000   
4                                        24000000   
...                                           ...   
1476                                     18000000   
1477                                     25000000   
1478                                    687000000   
1479                                      1000000   
1480                                      1000000   

      Military expenditure (current USD)  Military expenditure (% of GDP)  
0                           2.195802e+08                         2.566267  
1                           2.032433e+09                         3.114054  
2                           1.948403e+08                         1.820765  
3                           8.460994e+09                         3.280521  
4                           2.296448e+09                         0.792548  
...                                  ...                              ...  
1476                        5.924201e+09                         4.129198  
1477                        1.163608e+09                         2.321460  
1478                        7.782320e+11                         3.741160  
1479                        7.896501e+07                         1.060553  
1480                        2.121424e+08                         1.154225  

[1481 rows x 10 columns]

In [106]:
df.describe()

Unnamed: 0         Year  Arms imports (SIPRI trend indicator values)  \
count  1481.000000  1481.000000                                 1.481000e+03   
mean   1143.244429  2013.547603                                 2.418325e+08   
std     651.580640     3.973467                                 4.949061e+08   
min       0.000000  2007.000000                                 0.000000e+00   
25%     588.000000  2010.000000                                 1.000000e+07   
50%    1142.000000  2014.000000                                 4.800000e+07   
75%    1709.000000  2017.000000                                 2.300000e+08   
max    2283.000000  2020.000000                                 5.381000e+09   

       Military expenditure (current USD)  Military expenditure (% of GDP)  
count                        1.481000e+03                      1481.000000  
mean                         1.561759e+10                         2.121926  
std                          6.920290e+10                         1.677999  
min                          0.000000e+00                         0.000000  
25%                          4.015291e+08                         1.131309  
50%                          2.279622e+09                         1.594822  
75%                          6.846249e+09                         2.578160  
max                          7.782320e+11                        13.325672

# **IN 2017**

In [107]:
#Selecting the year
df1=df[df['Year']==2017]
df1

Unnamed: 0  Year Country Code   Country Name  \
1060        1634  2017          AFG    Afghanistan   
1061        1635  2017          AGO         Angola   
1062        1636  2017          ALB        Albania   
1063        1638  2017          ARG      Argentina   
1064        1639  2017          ARM        Armenia   
...          ...   ...          ...            ...   
1160        1786  2017          UGA         Uganda   
1161        1787  2017          UKR        Ukraine   
1162        1789  2017          USA  United States   
1163        1791  2017          VNM        Vietnam   
1164        1795  2017          ZMB         Zambia   

                          Region         Income Group    Lending Type  \
1060                  South Asia           Low income             IDA   
1061         Sub-Saharan Africa   Lower middle income            IBRD   
1062       Europe & Central Asia  Upper middle income            IBRD   
1063  Latin America & Caribbean   Upper middle income            IBRD   
1064       Europe & Central Asia  Upper middle income            IBRD   
...                          ...                  ...             ...   
1160         Sub-Saharan Africa            Low income             IDA   
1161       Europe & Central Asia  Lower middle income            IBRD   
1162               North America          High income  Not classified   
1163         East Asia & Pacific  Lower middle income            IBRD   
1164         Sub-Saharan Africa            Low income             IDA   

      Arms imports (SIPRI trend indicator values)  \
1060                                    252000000   
1061                                    110000000   
1062                                      4000000   
1063                                     47000000   
1064                                     18000000   
...                                           ...   
1160                                     21000000   
1161                                      4000000   
1162                                    516000000   
1163                                    783000000   
1164                                     44000000   

      Military expenditure (current USD)  Military expenditure (% of GDP)  
1060                        1.914071e+08                         0.945227  
1061                        3.062873e+09                         2.507985  
1062                        1.443827e+08                         1.108696  
1063                        5.459644e+09                         0.856138  
1064                        4.436104e+08                         3.849731  
...                                  ...                              ...  
1160                        3.467629e+08                         1.096079  
1161                        3.246787e+09                         2.884402  
1162                        6.467530e+11                         3.313381  
1163                        5.073854e+09                         2.267970  
1164                        3.396645e+08                         1.309596  

[105 rows x 10 columns]

In [108]:
#Selecting the intrested countries After reasearch
df2=df1[df1['Country Name'].isin(['United States','China','Russian Federation','India','Pakistan'])]
df2

Unnamed: 0  Year Country Code        Country Name  \
1079        1662  2017          CHN               China   
1103        1700  2017          IND               India   
1137        1748  2017          PAK            Pakistan   
1145        1759  2017          RUS  Russian Federation   
1162        1789  2017          USA       United States   

                     Region         Income Group    Lending Type  \
1079    East Asia & Pacific  Upper middle income            IBRD   
1103             South Asia  Lower middle income            IBRD   
1137             South Asia  Lower middle income           Blend   
1145  Europe & Central Asia  Upper middle income            IBRD   
1162          North America          High income  Not classified   

      Arms imports (SIPRI trend indicator values)  \
1079                                   1338000000   
1103                                   2909000000   
1137                                    837000000   
1145                                     93000000   
1162                                    516000000   

      Military expenditure (current USD)  Military expenditure (% of GDP)  
1079                        2.104430e+11                         1.746455  
1103                        6.455944e+10                         2.531463  
1137                        1.146125e+10                         3.774449  
1145                        6.691303e+10                         4.248996  
1162                        6.467530e+11                         3.313381

In [109]:
# plotting the bar graph for milatary expenditure in 2017
px.bar(df2, x = 'Country Name', y = 'Military expenditure (current USD)',color='Country Name',title='Bar chart shows how much countries are spending on their Military expenditure (in terms of USD)',height=500,
template=list(plotly.io.templates.keys())[5])

**OBSERVATIONS :**
   * In 2017, the United States' military spending (in current dollars) is nearly [$646 billion](#theme-installion). When it comes to China  its military expenditure is [210 billion](#theme-installion).          India's military expenditure (current USD) is [64.5 billion](#theme-installion), and Russia's military expenditure (current USD) is [66.9 billion](#theme-installion). Finally, Pakistan's military         expenditure in current USD is [11.4 billion](#theme-installion).

* Here, the U.S Military expenditure is at the top with $64Billion  and PAKISTHAN is at the bottom with 11.4Billion


  


In [110]:
# plotting the pie chart for arms import (SIPRI trend indicator values) Pie chart
px.pie(df2,values='Arms imports (SIPRI trend indicator values)',color='Country Name',names='Country Name',
    height=500,
    width=1000,
    template=list(plotly.io.templates.keys())[5],title='Arms imports( SIPRI trend indicator values) Pie Chart')

**OBSERVATIONS :**

  * In 2017, India has the most imports of any country at 51.1%, followed by China at 23.5%, Pakistan at 14.7%, the US at 9.06%, and Russia at 1.63%.

  * Here, the INDIA imports are at the top with 51.1% and RUSSIA is at the bottom with 1.63%.

In [111]:
# Selecting the required countries 
df3 = df[df['Country Name'].isin(['United States','China','Russian Federation','India','Pakistan'])]
df3

Unnamed: 0  Year Country Code        Country Name  \
19            28  2007          CHN               China   
40            67  2007          IND               India   
69           114  2007          PAK            Pakistan   
76           126  2007          RUS  Russian Federation   
93           153  2007          USA       United States   
...          ...   ...          ...                 ...   
1384        2115  2019          USA       United States   
1407        2150  2020          CHN               China   
1427        2187  2020          IND               India   
1458        2236  2020          PAK            Pakistan   
1478        2277  2020          USA       United States   

                     Region         Income Group    Lending Type  \
19      East Asia & Pacific  Upper middle income            IBRD   
40               South Asia  Lower middle income            IBRD   
69               South Asia  Lower middle income           Blend   
76    Europe & Central Asia  Upper middle income            IBRD   
93            North America          High income  Not classified   
...                     ...                  ...             ...   
1384          North America          High income  Not classified   
1407    East Asia & Pacific  Upper middle income            IBRD   
1427             South Asia  Lower middle income            IBRD   
1458             South Asia  Lower middle income           Blend   
1478          North America          High income  Not classified   

      Arms imports (SIPRI trend indicator values)  \
19                                     1639000000   
40                                     2275000000   
69                                      694000000   
76                                       90000000   
93                                      832000000   
...                                           ...   
1384                                    866000000   
1407                                    811000000   
1427                                   2799000000   
1458                                    759000000   
1478                                    687000000   

      Military expenditure (current USD)  Military expenditure (% of GDP)  
19                          6.213659e+10                         1.739813  
40                          2.825477e+10                         2.478248  
69                          5.342575e+09                         3.511986  
76                          4.353499e+10                         3.118542  
93                          5.895860e+11                         4.079655  
...                                  ...                              ...  
1384                        7.343440e+11                         3.427080  
1407                        2.523040e+11                         1.749796  
1427                        7.288745e+10                         2.882670  
1458                        1.037638e+10                         4.024714  
1478                        7.782320e+11                         3.741160  

[67 rows x 10 columns]

In [112]:
#plotting line chart for Military expenditure from 2007 to 2017 on selected countries
px.line(df3,x='Year',y='Military expenditure (current USD)',color='Country Name',height=500,
    width=1000,
    template=list(plotly.io.templates.keys())[5],title='The below line chart shows the trend for 2007 to 2020 on Military expenditure(current USD)')

**OBSERVATIONS :**
   * **UNITED STATES** The line chart above shows how much money was spent on the military (in current USD) from [2007 to 2020](#theme-installion). As time goes from left to right, the United States has always been on top when comparing from all selected the countries. In [2007](#theme-installion), it spent [589.5 billion dollars](#theme-installion), and in [2011](#theme-installion), it spent [752 billion dollars](#theme-installion), which was the most ever. From [2012 to 2018](#theme-installion), the amount spent by the US slowly went down to about [682B](#theme-installion). In [2020](#theme-installion), it will be almost [778B](#theme-installion), which is the most the country has ever spent.
   
   * **CHINA** The line graph below depicts China's total military expenditures (in constant USD) from [2007 to 2020](#theme-installion). In the early stages, China spent [62.5 billion dollars](#theme-installion) from there, its spending gradually increased, and by [2016](#theme-installion), it had reached the level it had maintained in [2015: a total of 196.5 billion dollars](#theme-installion) and now the country that has spent the most money overall, at a total of [252.3 billion dollars](#theme-installion).
   
   * **RUSSIAN FEDERATION** From the line graph, you can see how much Russia spent on its military (in current USD) from [2007 to 2020](#theme-installion). As time goes from left to right, Russia has been in the [3 rd place](#theme-installion) among the selected countries. In [2007, it spent $62.1 billion](#theme-installion). Since then, the spending has gone up and down a bit, but between [2017 and 2020](#theme-installion), it dropped faster than India and spent nearly [65.6 billion dollars](#theme-installion). It kept going until [2020](#theme-installion) and ended up in [4th place](#theme-installion).
   
   * **INDIA** From [2007 to 2020](#theme-installion), the line graph shows how much India has spent on its military (in current USD). As you move from left to right, you can see that India was in [4 th place in 2007 with 28.4 billion](#theme-installion), and then it started to grow slowly until [2017](#theme-installion). But in [2018](#theme-installion), it spent more than Russia, with [66.8 billion, and by 2020](#theme-installion), it was in [3 rd place, with 72.8 billion](#theme-installion).

   * **PAKISTHAN** This line graph shows Pakistan's military spending from [2007 to 2020 (in current USD)](#theme-installion). As time goes on, from left to right, Pakistan has always been in [5th place](#theme-installion). In [2007, it has spent 5.32 billion](#theme-installion), and it stayed there until [2020, when it had 10.8 billion](#theme-installion).

In [113]:
#plotting the scatter plot for Relationship between Milatary expenditure (current USD) and Arms import (SIPRI trend indicator values)
px.scatter(df1,
    x="Military expenditure (current USD)",
    y="Arms imports (SIPRI trend indicator values)", 
    color='Country Name',
    height=500,
    width=1000,
    template=list(plotly.io.templates.keys())[5],
    title="Relationship between Milatary expenditure (current USD) and Arms import (SIPRI trend indicator values)")

**OBSERVATIONS :**
   * This scatterplot shows that there is a direct relationship between Military spending (current USD) and Arms imports (SIPRI trend indicators values) across all countries. Out of all the countries, Saudi Arabia spent the most on importing arms, spending 105 billion dollars to buy 4.2 billion arms. India spends 100 billion dollars to bring in 3.2 billion arms. But China has spent 220 billion dollars on imports, which is less than SAU and IND. At the same time, the United States has spent 650 billion dollars on imports, but it only bought 0.7 billion arms and came in last place.
   
   * This scatterplot shows countries who are investing more they have very less imports for fire arms but especially a few countries like IND and SAU, EGY and AUS.

In [114]:
#plotting the scatter plot for Relationship between Milatary expenditure (% of GDP) and Milatary expenditure (current USD)
px.scatter(df1,
    x="Military expenditure (% of GDP)",
    y="Military expenditure (current USD)", 
    color='Country Name',
    height=500,
    width=1000,
    template=list(plotly.io.templates.keys())[5],
    title="Relationship between Milatary expenditure (% of GDP) and Milatary expenditure (current USD)")

**OBSERVATIONS :**

* This scatterplot shows the relationship between military spending (as a percentage of GDP) and military spending (in current dollars), such as how much of their money is spent on the military. but SAU and OMN spend a higher percentage of their budgets on the military. If they could put money into other areas, like education and health care, those areas could also get better.

# Conclusion
The military budget is a big part of a country's overall budget. From an economic point of view, military spending is part of what the government spends on national defense. Most countries don't want to import a lot of arms if they have good technologies. Also, countries that invest a lot need to keep an eye on their military spending because they could invest in other things instead.

